src/IncludedCommands/Administrative/slock.lua

```diff
@@ -13,44 +13,89 @@ local Types = require(script.Parent.Parent.Parent:WaitForChild("Types"))
                 for _, Admin in Players:GetPlayers() do
-                    if Api.Authorization:GetAdminLevel(Admin) >= 0 then
+                    if Api.Authorization:GetAdminLevel(Admin) >= Api.CommandData.ServerLock.AdminLevel then
                         Api.Messages:DisplayHint(Admin, Player.Name.." ("..Player.DisplayName..", "..tostring(Player.UserId)..") tried to enter the server.")
                     end
                 end
             end
         end)
+        Players.PlayerRemoving:Connect(function()
+            -- Set HighestAdminLevel to an unreasonably low number to avoid error
+            local HighestAdminLevel = -math.huge
+            for _, Player in Players:GetPlayers() do
+                local AdminLevel = Api.Authorization:YieldForAdminLevel(Player)
+                if AdminLevel > HighestAdminLevel then
+                    HighestAdminLevel = AdminLevel
+                end
+            end
+            
+            -- If there is no player with a higher AdminLevel than the current ServerLock AdminLevel disable ServerLock
+            if HighestAdminLevel < Api.CommandData.ServerLock.AdminLevel then
+                Api.CommandData.ServerLock = {
+                    Locked = false,
+                    AdminLevel = 0,    
+                }
+            end
+        end)
     end,
-    ServerRun = function(CommandContext: Types.CmdrCommandContext, Active: boolean?)
+    ServerRun = function(CommandContext: Types.CmdrCommandContext, Active: boolean?, AdminLevel: number?)
         local Util = IncludedCommandUtil.ForContext(CommandContext)
         local Api = Util:GetApi()
+        
+        print(Api.CommandData.ServerLock)
+        
+        if Api.Authorization:YieldForAdminLevel(CommandContext.Executor) < Api.CommandData.ServerLock.AdminLevel then
+            return "Cannot change ServerLock above your own AdminLevel."
+        end
 
         --Set the server lock.
         if Active ~= nil then
-            Api.CommandData.ServerLocked = Active
+            Api.CommandData.ServerLock.Locked = Active
+
+            --Set the server lock AdminLevel
+            if AdminLevel ~= nil then
+                if Api.Authorization:YieldForAdminLevel(CommandContext.Executor) < AdminLevel then
+                    return "Cannot set ServerLock above your own AdminLevel."
+                end
+                Api.CommandData.ServerLock.AdminLevel = AdminLevel
+            end
         else
-            Api.CommandData.ServerLocked = not Api.CommandData.ServerLocked
+            if Api.Authorization:YieldForAdminLevel(CommandContext.Executor) < Api.Configuration.DefaultAdminLevel + 1 then
+                return "Cannot set ServerLock above your own AdminLevel."
+            end
+            Api.CommandData.ServerLock.Locked = not Api.CommandData.ServerLock.Locked
+            Api.CommandData.ServerLock.AdminLevel = Api.Configuration.DefaultAdminLevel + 1
```

Comment:
`Api.Configuration.DefaultAdminLevel + 1` is *not* a safe assumption to go with. `DefaultAdminLevel` is the admin level given to players by default (typically <0), not the minimum level to be considered an admin (currently 0). At least 1 game I have worked on used multiple admin levels below 0 for player-facing functionality. For when the value is changed, this would make `slock` not default to 0.